hr_holidays_imposed_days/models/hr_imposed_holidays.py

```diff
@@ -0,0 +1,120 @@
+                                    help="If empty, all employees of the "
+                                         "company will have this day/period "
+                                         "imposed")
+    auto_confirm = fields.Boolean()
+
+    @api.multi
+    def validate(self):
+        for rec in self:
+            created = self.env['hr.holidays']
+            if rec.employee_ids:
+                employees = rec.employee_ids
+            else:
+                employees = self.env['hr.employee'].search(
+                    [('company_id', '=', rec.company_id.id)])
+
+            for employee in employees:
+                vals = {
+                    'number_of_days_temp': 1.,
+                    'name': rec.name,
+                    'date_from': rec.date_from,
+                    'date_to': rec.date_to,
+                    'employee_id': employee.id,
+                    'type': 'remove',
+                    'holiday_status_id': rec.status_id.id,
+                }
+
+                created |= created.create(vals)
+            if rec.auto_confirm:
+                created.signal_workflow('validate')
+
+    def _get_number_of_days(self, date_from, date_to):
+        """Returns a float equals to the timedelta between
+           two dates given as string."""
+
+        DATETIME_FORMAT = tools.DEFAULT_SERVER_DATETIME_FORMAT
+        from_dt = datetime.datetime.strptime(date_from, DATETIME_FORMAT)
+        to_dt = datetime.datetime.strptime(date_to, DATETIME_FORMAT)
+        timedelta = to_dt - from_dt
+        diff_day = timedelta.days + float(timedelta.seconds) / 86400
+        return diff_day
+
+    @api.onchange('date_from', 'date_to')
+    def onchange_date_from(self):
+        """
+        If there are no date set for date_to, automatically set one
+        8 hours later than the date_from.
+        Also update the number_of_days.
+        """
+        # date_to has to be greater than date_from
+        if ((self.date_from and self.date_to) and
+                (self.date_from > self.date_to)):
+            raise exceptions.Warning(
+                _('The start date must be anterior to the end date.'))
+
+        # No date_to set so far: automatically compute one 8 hours later
+        if self.date_from and not self.date_to:
+            date_to_with_delta = datetime.datetime.strptime(
+                self.date_from,
+                tools.DEFAULT_SERVER_DATETIME_FORMAT) + datetime.timedelta(
+                hours=8)
+            self.date_to = str(date_to_with_delta)
+
+        # Compute and update the number of days
+        if ((self.date_to and self.date_from) and
+                (self.date_from <= self.date_to)):
+            diff_day = self._get_number_of_days(self.date_from, self.date_to)
+            self.number_of_days = round(math.floor(diff_day))+1
+        else:
+            self.number_of_days = 0
+
+    @api.onchange('date_from', 'date_to')
+    def onchange_date_to(self):
```

Comment:
The two onchange methods are both triggered on the same fields.
I think you should trigger each method on a single field, or merge the two methods.